packages/wonder-blocks-modal/components/modal-header.js

```diff
@@ -4,38 +4,67 @@ import {StyleSheet} from "aphrodite";
 import Color from "@khanacademy/wonder-blocks-color";
 import {View} from "@khanacademy/wonder-blocks-core";
 import {MediaLayout} from "@khanacademy/wonder-blocks-layout";
-
-import type {StyleType} from "@khanacademy/wonder-blocks-core";
+import {
+    HeadingMedium,
+    LabelMedium,
+} from "@khanacademy/wonder-blocks-typography";
 
 type Props = {|
-    children: React.Node,
-    style?: StyleType,
-    color: "light" | "dark",
+    /**
+     * The main title rendered in larger bold text.
+     */
+    title: string,
+
+    /**
+     * The dialog subtitle.
+     */
+    subtitle?: string,
+
+    /**
+     * Adds a breadcrumb-trail, appearing in the ModalHeader, above the title.
+     */
+    breadcrumbs?: React.Node,
+
+    /**
+     * Whether to display the "light" version of this component instead, for
+     * use when the item is used on a dark background.
+     */
+    light: boolean,
 |};
 
 export default class ModalHeader extends React.Component<Props> {
     static defaultProps = {
-        color: "dark",
+        light: true,
     };
 
-    static __IS_MODAL_HEADER__ = true;
-    static isClassOf(instance: any) {
-        return instance && instance.type && instance.type.__IS_MODAL_HEADER__;
-    }
-
     render() {
-        const {style, color, children} = this.props;
+        const {light, title, subtitle, breadcrumbs} = this.props;
+
+        if (subtitle && breadcrumbs) {
```

Comment:
@kevinbarabash @jeresig At the end I decided to go with this approach b/c Flow threw an error when trying to render these props inside the `render` method. You can see the example here:

https://flow.org/try/#0PTAEAEDMBsHsHcBQBLAtgB1gJwC6gFSgCGAzqAEoCmRAxnpFrKqAERbV0sDcivOAnukqgAQqUoAFRujIBeUAG8APolBrQNWADsclHQC4KHHADoAcrAAmlADSr1OZDmiVDJHFmRaA5nfWhIWFhdLAB+QypaUwtrP3VtAGE4Ekpw0AAKAEpQWQA+UFRkAA9KSzi1IgAjWAA3VIjjcytbezVKyjh4NMi6JtjW0F13AElLNPdPHzslAF8eRAEhUAB1JwALAGUAV0rHZ2F5ZQGTE7EUqVgZctASHb2XNw8vX0RZ+cXhVZw1kXYiSxoWC2qEqckUKn8JxMZ0k0hI10qfwBQJBJAaUT6LTefEEwguMhyonE+LIShW622uycLlAZK+PyRgOBoPmiBo0FIZAAslYiNAABLUaxYUCUIq6LSWMg9UwJJiYLR6HAAHhJ+QU9nYksoWCyag1-nYOC2WC0oGVawAjLkFN9kCQTOg4SZ7pQZoo7Q6nZcHYihUzUTNlcArbkeGoZohI2ztO5QGshTrLYTlTzLHzBf8dRptBKcLIWEMcCxBtTKAXArASyRLJAC7XLCXgGGY1o4wms1gAEwptMZxMizQ6JUFosl10VoLVu5lguVIhYJstoftgcAZl7vIFA5zw50o8o7nHs5YlZLfv+AdBsgUAG0WPOAF4sGysACOWyKLAAuu7mzwVzwDthQAFk3dNt07Xc8wPI9S32Scq1AC9kWZEgb3vJ8X3fT8f3dW4qQQh8FyXHggA